addons/im_livechat/static/src/public_models/public_livechat_global.js

```diff
@@ -132,6 +144,29 @@ registerModel({
                 this.chatbot.restoreSession();
             }
         },
+
+        getVisitorUserId() {
+            const cookie = JSON.parse(getCookie("im_livechat_session") || "{}");
+            if ("visitor_uid" in cookie) {
+                return cookie.visitor_uid;
+            }
+            return session.user_id;
+        },
+
```

Comment:
Ok as a fix in stable. For master (can be follow-up PR), we could keep the cookie in RAM, so that we don't have to read the same cookie several times (cookie reading and parsing is quite slow)

Reply:
I'm not sure its worth it:

![image](https://github.com/odoo/odoo/assets/48757558/09eaa52e-57d2-4257-97bb-b30716cb712e)